posts/fantastic-simprocs.md

```diff
@@ -0,0 +1,176 @@
+---
+author: 'Yaël Dillies, Paul Lezeau'
+category: 'Metaprogramming'
+date: 2025-03-29 12:33:00 UTC+00:00
+description: 'Brief introduction to simprocs and what they are useful for'
+has_math: true
+link: ''
+slug: fantastic-simprocs
+tags: 'simp, simproc, meta'
+title: Fantastic Simprocs and Where to Find Them.
+type: text
+---
+
+Lean core added support for custom simplification procedures, aka *simprocs*.
+This blog post is the first in a series of two aimed at explaining what a simproc is, what kind of problems can be solved with simprocs, and what tools we have to write them.
+
+<!-- TEASER_END -->
+
+This post describes purely informally what simprocs are and do.
+The second post will be a walkthrough to writing a simple simproc in three different ways.
+
+# What is a simproc
+
+To understand what a simproc is and how it works, we will first explain how `simp` works.
+Then we will give some examples and non-examples of simprocs as well as pointers to analogous concepts in other theorem provers.
```

Comment:
Perhaps I'm missing something, but I don't actually see an explanation of what simprocs do here! You explain what simp does, then have a single sentence giving one intuition for what a simproc is, but you don't say what it actually does. And then you jump to the examples. You also mention non-examples and pointers to other theorem provers, but I don't see that in this post either?

Reply:
There are now several sentences. I'm not necessarily happy with the amount of explanations though